src/inc/eventtracebase.h

```diff
@@ -168,24 +175,102 @@ class Object;
 #define TRACE_LEVEL_INFORMATION 4   // Includes non-error cases such as Entry-Exit
 #define TRACE_LEVEL_VERBOSE     5   // Detailed traces from intermediate steps
 
-struct ProfilingScanContext;
+//
+// Using KEYWORDZERO means when checking the events category ignore the keyword
+//
+#define KEYWORDZERO 0x0
 
 //
-// Use this macro to check if ETW is initialized and the event is enabled
+// Keywords macro defintitions for cross-platforms
+// On Windows, clretwall.h gets generated and included.
+// On Linux/macOS, it isn't. So we define these keywords in here.
 //
-#define ETW_TRACING_ENABLED(Context, EventDescriptor) \
-    ((Context.IsEnabled && ETW_TRACING_INITIALIZED(Context.RegistrationHandle) && ETW_EVENT_ENABLED(Context, EventDescriptor)) || EVENT_PIPE_ENABLED())
+#ifdef FEATURE_PAL
+//
+// CLR Provider Keywords
+//
+#define CLR_GC_KEYWORD 0x1
```

Comment:
I'm hoping we could produce these constants using codegen from the manifest with our python script. Otherwise we have an awkward synchronization issue where developers need to update this list and the manifest on each change.
Let me know if there are obstacles preventing that.